common/util_test.go

```diff
@@ -21,3 +23,42 @@ func TestTxDataToVideoProfile(t *testing.T) {
 		t.Error("Unexpected profile! ", err, res)
 	}
 }
+
+func TestVideoProfileBytes(t *testing.T) {
+	if len(VideoProfileByteLookup) != len(VideoProfileNameLookup) {
+		t.Error("Video profile byte map was not created correctly")
+	}
+	if res, err := BytesToVideoProfile(nil); err != nil && len(res) != 0 {
+		t.Error("Unexpected return on empty input")
+	}
+	if res, err := BytesToVideoProfile([]byte{}); err != nil && len(res) != 0 {
+		t.Error("Unexpected return on empty input")
+	}
+	if _, err := BytesToVideoProfile([]byte("abc")); err != ErrProfile {
+		t.Error("Unexpected return on too-short input", err)
+	}
+	if _, err := BytesToVideoProfile([]byte("abcdefghijk")); err != ErrProfile {
+		t.Error("Unexpected return on invalid input", err)
+	}
+	b, _ := hex.DecodeString("93c717e7c0a6517a")
+	res, err := BytesToVideoProfile(b)
+	if err != nil || res[1] != ffmpeg.P240p30fps16x9 || res[0] != ffmpeg.P360p30fps16x9 {
+		t.Error("Unexpected profile! ", err, res)
+	}
+}
+
+func TestProfilesToHex(t *testing.T) {
+	// Sanity checking against an existing eth impl that we know works
+	compare := func(profiles []ffmpeg.VideoProfile) {
+		b1 := ethcommon.ToHex(ProfilesToTranscodeOpts(profiles))[2:]
+		b2 := ProfilesToHex(profiles)
+		if b1 != b2 {
+			t.Error("Unequal profile hex ")
+		}
+	}
+	// XXX double check which one is wrong! ethcommon method produces "0" zero string
```

Comment:
And the reason for using `hex.EncodeToString` over `ethcommon.ToHex`within `ProfilesToHex` is because the latter appends a leading `0x`, which we end up removing nearly every time anyway, even among our other uses of `ethcommon.Hex`.

So here we avoid the extra manipulation and use a function that gives us the results we want.

Reply:
Ahh I see.  I guess technically `ethcommon.Hex` is in context of Ethereum, and in that world `0x` might be fine, but I agree it's confusing.  I'm fine with either as long as we are wrapping that so programmers don't have to think about it.

Reply:
`ethcommon.ToHex` appends a leading `0x`, which when removed, which equals the output of `hex.EncodeToString`? What do you mean by wrapping? Do programmers have to "think about it" the way it's written here? Just trying to understand what's going on here!

Reply:
> ethcommon.ToHex appends a leading 0x, which when removed, which equals the output of hex.EncodeToString?

Correct, in most cases. What's under discussion here is the cases where the outputs are *not* equal between the two hex-encoding functions.

> What do you mean by wrapping? Do programmers have to "think about it" the way it's written here?

Many of the uses of `ToHex` throughout the codebase involve stripping the leading "0x" prefix with `ethcommon.ToHex(string)[2:]`

Wrapping means the programmer doesn't have "think about" doing that stripping, regardless of the hex-encoding method that's being used. It is a form of encapsulation. Here, we use `hex.EncodeToString` in `ProfilesToHex` just so we don't have to strip at all, but even if we used `ethcommon.ToHex` instead, its use would be wrapped by the `ProfilesToHex` function, so the programmer doesn't have to be aware of the differences in hex-encoding methods.

Reply:
That makes sense. I do see where it's being used this way `ethcommon.ToHex(string)[2:]`. I understand now, thank you Josh!